pvlib/iotools/tmy.py

```diff
@@ -519,3 +519,179 @@ def _read_tmy2(string, columns, hdr_columns, fname):
+    liquid_precipitation_quantity       The period of accumulation for the liquid precipitation depth field, hour
+    =============================       ======================================================================================================================================================
+
+    References
+    ----------
+
+    [1] EnergyPlus documentation, Auxiliary Programs
+    https://energyplus.net/documentation.
+    '''
+    
+    if filename is None:
+        try:
+            filename = _interactive_load()
+        except ImportError:
+            raise ImportError('Interactive load failed. Tkinter not supported '
+                              'on this system. Try installing X-Quartz and '
+                              'reloading')
+    
+    if filename.startswith('http'):
+        # Attempts to download online EPW file
+        # See comments above for possible online sources
+        request = Request(filename, headers={'User-Agent': (
+            'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) '
+            'AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 '
+            'Safari/537.36')})
+        response = urlopen(request)
+        csvdata = io.StringIO(response.read().decode(errors='ignore'))
+    else:
+        # Assume it's accessible via the file system
+        csvdata = open(filename, 'r')
+    
+    # Read line with metadata
+    firstline = csvdata.readline()
+
+    head = ['loc','city', 'state-prov', 'country', 'data_type','WMO_code', 
+            'latitude', 'longitude', 'TZ','altitude']
+    meta = dict(zip(head, firstline.rstrip('\n').split(",")))
+
+    meta['altitude'] = float(meta['altitude'])
+    meta['latitude'] = float(meta['latitude'])
+    meta['longitude'] = float(meta['longitude'])
+    meta['TZ'] = float(meta['TZ'])
+
+    colnames = ['year', 'month', 'day', 'hour', 'minute', 'data_source_unct', 
+                't_drybulb', 't_dewpoint', 'rel_hum', 'atmospheric_pressure',
+                'etr', 'etrn', 'ghi_infrared', 'ghi', 'dni', 'dhi',
+                'global_hor_illum', 'direct_normal_illum', 
+                'diffuse_horizontal_illum', 'zenith_luminance', 
+                'wind_direction', 'wind_speed', 'total_sky_cover', 
+                'opaque_sky_cover', 'visibility', 'ceiling_height', 
+                'present_weather_observation', 'present_weather_codes',
+                'precipitable_water', 'aerosol_otpical_depth', 'snow_depth', 
+                'days_since_last_snowfall', 'albedo', 
+                'liquid_precipitation_depth', 'liquid_precipitation_quantity']
+   
+    # We only have to skip 6 rows instead of 7 because we have already used 
+    # the realine call above.
+    data = pd.read_csv(csvdata, skiprows=6, header = 0, names=colnames)
+    
+    # Shift one hour back because EPW's usage of hour 24
+    # and dateutil's inability to handle that.  
+    data["hour"] = data["hour"].apply(lambda x: x - 1)
```

Comment:
The epw function in [this comment](https://github.com/pvlib/pvlib-python/issues/261#issuecomment-337231024) did not apply a time shift. Which is correct?

Reply:
Thanks. EPW data files use a similar 1-24h formulation as TMY3, so I mistakenly assumed this shift would be necessary. Apparently, `pd.to_datetime` was already handling things just fine, but in this update I had to add a correction to the last line to avoid violation of all dates belonging to the same year. There must be a prettier solution for this, but it seems to work for the time being.